source/renderer/app/components/wallet/settings/ChangeSpendingPasswordDialog.js

```diff
@@ -110,6 +110,22 @@ export default class ChangeSpendingPasswordDialog extends Component<Props> {
             messages.currentPasswordFieldPlaceholder
           ),
           value: '',
+          validators: [
+            ({ field }) => {
+              if (this.props.isSpendingPasswordSet) {
+                if (field.value !== '') {
```

Comment:
@DeeJayElly can you create one if clause

Reply:
We have 3 conditions here based on the this.props.isSpendingPasswordSet and field.value set or not set.